### ISSUE TITLE
NS1004 warning, when using static wrapper class

### ISSUE THREAD

**Issue by NPadrutt** [nsubstitute/NSubstitute.Analyzers#204]
I use mediator and wrap my queries with the handler in a static class like this:

```
public static class GetProjectDetails
{
    public record Query(Guid CustomerId, string ProjectNumber) : IRequest<ProjectDetailData>;

    public class Handler : IRequestHandler<Query, ProjectDetailData>
    {
        private readonly IAppDbContext context;

        public Handler(IAppDbContext context)
        {
            this.context = context;
        }

        public async Task<ProjectDetailData> Handle(Query request, CancellationToken cancellationToken)
        {
            ...
        }
    }
}
```

When I now substitute this in my test with this, I get a NS1004 warning, which I think is a false positive: 

`mediator.Send(Arg.Any<GetProjectDetails.Query>())`

Same is with the Received call for that matter, although here I also get a NS5000 in addition: 

`await mediator.Received(1).Send(Arg.Any<GetProjectDetails.Query>());`

**Comment by tpodolak** [comment]
Hi @NPadrutt 
I am not able to reproduce this with [newest](https://www.nuget.org/packages/NSubstitute.Analyzers.CSharp/1.0.16) analyzers. Here is my code based on your comment
```c#
using NSubstitute;
using System;
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Xunit;

namespace MyNamespace
{
    public class FooTests
    {
        [Fact]
        public void Test()
        {
            var substitute = NSubstitute.Substitute.For<IMediator>();
            substitute.Received(1).Send(Arg.Any<GetProjectDetails.Query>());
        }
    }
}

public static class GetProjectDetails
{
    public record Query(Guid CustomerId, string ProjectNumber) : IRequest<ProjectDetailData>;

    public class Handler : IRequestHandler<Query, ProjectDetailData>
    {
        private readonly IAppDbContext context;

        public Handler(IAppDbContext context)
        {
            this.context = context;
        }

        public Task<ProjectDetailData> Handle(Query request, CancellationToken cancellationToken)
        {
            return Task.FromResult<ProjectDetailData>(null);
        }
    }
}

public interface IAppDbContext
{
}

public class ProjectDetailData
{
}
```
Is mediator variable in your case `IMediator` interface or `Mediator` class?

**Comment by NPadrutt** [comment]
hey @tpodolak 
thanks for looking into this. You are right, the error is no longer showing up. Maybe that was a hiccup of Rider. The NS5000  is still showing, but I'll create a separate issue for that if it persists.
